test/test.js

```diff
@@ -172,10 +176,10 @@ test('folded headers', function(t) {
   var recp = 'bar@gmail.com', subject = 'email test why not', body = 'Why not a test email?',
   xfolded = 'This is a\r\n folded header', xfoldedtab = 'This is a\r\n\t tab folded header',
   handler = function(address, id, email) {
-    t.equal(email.headers.To,         recp);
-    t.equal(email.headers.From,       sender);
-    t.equal(email.headers.Xfolded,    xfolded.replace(/[\r\n\t]/g,''));
-    t.equal(email.headers.Xfoldedtab, xfoldedtab.replace(/[\r\n\t]/g,''));
+    t.equal(email.headers.To,                recp);
+    t.equal(email.headers.From,              sender);
+    t.equal(email.headers.get('xfolded'),    'This is a  folded header');
+    t.equal(email.headers.get('xfoldedtab'), 'This is a \t tab folded header');
```

Comment:
It's better to see the actual value expected as a literal instead of having to run some code in your head to know what the expected value is.

I changed this because this is how MailParser was behaving, but after your comment I went to read more about folding. If I understood correctly:

Newlines are generally not allowed in header values:

> A field body may be composed of printable US-ASCII characters
> as well as the space (SP, ASCII value 32) and horizontal tab (HTAB,
> ASCII value 9) characters (together known as the white space
> characters, WSP).  A field body MUST NOT include CR and LF except
> when used in "folding" and "unfolding", as described in section
> 2.2.3.

> Each header field is logically a single line of characters comprising
> the field name, the colon, and the field body.  For convenience
> however, and to deal with the 998/78 character limitations per line,
> the field body portion of a header field can be split into a
> multiple-line representation; this is called "folding".  The general
> rule is that wherever this specification allows for folding white
> space (not simply WSP characters), a CRLF may be inserted before any
> WSP.

So folding is a convenience for when you read email in your 80 char terminal. I tried it and in fact Nodemailer folds header values that are too long; MailParser unfolds them.

I think the correct test here would be to provide a really long value, and make sure it is parsed correctly into a single line.